pkg/idtools/idtools_test.go

```diff
@@ -273,3 +276,155 @@ func TestIsContiguous(t *testing.T) {
+		expectedMinor int
+	}{
+		{
+			name:         "valid regular file",
+			input:        []byte("1000:1001:0644:file"),
+			expectErr:    false,
+			expectedUID:  1000,
+			expectedGID:  1000,
+			expectedMode: 0o644,
+		},
+		{
+			name:         "valid directory",
+			input:        []byte("1000:1001:0755:dir"),
+			expectErr:    false,
+			expectedUID:  1000,
+			expectedGID:  1000,
+			expectedMode: 0o755 | os.ModeDir,
+		},
+		{
+			name:      "Invalid format: missing fields",
+			input:     []byte("1000:1001"),
+			expectErr: true,
+		},
+		{
+			name:      "Invalid UID format",
+			input:     []byte("invalid:1001:0644:file"),
+			expectErr: true,
+		},
+		{
+			name:         "valid symlink",
+			input:        []byte("1000:1001:0777:symlink"),
+			expectErr:    false,
+			expectedUID:  1000,
+			expectedGID:  1000,
+			expectedMode: 0o777 | os.ModeSymlink,
+		},
+		{
+			name:         "valid pipe",
+			input:        []byte("1000:1001:0666:pipe"),
+			expectErr:    false,
+			expectedUID:  1000,
+			expectedGID:  1000,
+			expectedMode: 0o666 | os.ModeNamedPipe,
+		},
+		{
+			name:          "valid block device",
+			input:         []byte("1000:1001:0600:block-5-6"),
+			expectErr:     false,
+			expectedUID:   1000,
+			expectedGID:   1000,
+			expectedMode:  0o600 | os.ModeDevice,
+			expectedMajor: 5,
+			expectedMinor: 6,
+		},
+		{
+			name:          "valid char device",
+			input:         []byte("1000:1001:0600:char-10-11"),
+			expectErr:     false,
+			expectedUID:   1000,
+			expectedGID:   1000,
+			expectedMode:  0o600 | os.ModeCharDevice | os.ModeDevice,
+			expectedMajor: 10,
+			expectedMinor: 11,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			stat, err := parseOverrideXattr(tt.input)
+			if tt.expectErr {
+				assert.Error(t, err)
+			} else {
+				assert.NoError(t, err)
+				assert.NotNil(t, stat)
+				assert.Equal(t, tt.expectedUID, stat.IDs.UID)
+				assert.Equal(t, tt.expectedGID, stat.IDs.GID)
+				assert.Equal(t, tt.expectedMode, stat.Mode)
+				assert.Equal(t, tt.expectedMajor, stat.Major)
+				assert.Equal(t, tt.expectedMinor, stat.Minor)
+			}
+		})
+	}
+}
+
+func TestFormatContainersOverrideXattrDevice(t *testing.T) {
+	tests := []struct {
+		name     string
+		uid      int
+		gid      int
+		mode     int
+		major    int
+		minor    int
+		expected string
+	}{
+		{"regular file", 1000, 1001, 0o644, 0, 0, "1000:1001:0644:file"},
+		{"directory", 1000, 1001, 0o755 | int(os.ModeDir), 0, 0, "1000:1001:0755:dir"},
+		{"symlink", 1000, 1001, 0o777 | int(os.ModeSymlink), 0, 0, "1000:1001:0777:symlink"},
+		{"pipe", 1000, 1001, 0o666 | int(os.ModeNamedPipe), 0, 0, "1000:1001:0666:pipe"},
+		{"socket", 1000, 1001, 0o700 | int(os.ModeSocket), 0, 0, "1000:1001:0700:socket"},
+		{"block device", 1000, 1001, 0o600 | int(os.ModeDevice), 8, 0, "1000:1001:0600:block-8-0"},
+		{"char device", 1000, 1001, 0o600 | int(os.ModeDevice|os.ModeCharDevice), 1, 3, "1000:1001:0600:char-1-3"},
```

Comment:
Also test permission handling - at least `ISUID`/`ISGID`/`ISVTX`, or possibly each of the 12 bits.